lib/reline/ansi.rb

```diff
@@ -15,6 +15,7 @@ class Reline::ANSI
     'cud' => :ed_next_history,
     'cuf' => :ed_next_char,
     'cub' => :ed_prev_char,
+    'kdch1' => :key_delete 
```

Comment:
Hi @kcculhwch, I have another suggestion that makes sense to include in your PR:

The four lines above this for 'cuu', 'cud', 'cuf', and 'cub' don't belong and should be removed (along with the special handling for them on lines 100-101).  With your `smkx` fix in place, I don't think they'll be needed anymore.

I think what happened is @aycabta discovered that the cursor keys weren't working with some terminal emulators, including xterm, and he discovered that by mapping these it fixed it.

But that wasn't the right fix; the right fix is your PR, because as you and I both discovered independently, until the terminal has been set to 'smkx' mode, the home and end keys may send different escape sequences that don't align with the output of terminfo.  Well, **the same is true for the cursor keys**.

In other words, this list already includes 'kcuu1', 'kcud1', 'kcuf1', and 'kcub1', which are the correct things to map for the arrow keys.  They just weren't working (for some terminals) because reline wasn't putting the terminal in 'smkx' mode.  Now with your PR they will work.

The escape sequences for 'cuu', 'cud', 'cuf', and 'cub' are **not** something the terminal sends when arrow keys are pressed!  They are what you send **to** the terminal to *make the cursor move*.

It just so happens that in some terminals (well xterm at least), the escape sequences for  'cuu', 'cud', 'cuf', and 'cub' with the "%p1%d" removed from the middle happen to result in the escape sequences for the arrow keys in normal mode, but we don't need to rely on this happenstance anymore.